src/pivot.js

```diff
@@ -0,0 +1,228 @@
+  // If drilldown mode is enabled for columns or rows
+  let prevRowDrill = _.get('pagination.rows.drilldown', previous)
+  let prevColumnDrill = _.get('pagination.columns.drilldown', previous)
+
+  extend(node, {
+    pagination: {
+      columns: {
+        drilldown: prevColumnDrill && [],
+        skip: [],
+        expanded: [],
+      },
+      rows: {
+        drilldown: prevRowDrill && [],
+        skip: [],
+        expanded: [],
+      },
+    },
+  })
+}
+
+// Resetting the row pagination when the sorting is changed
+let resetExpandedRows = (extend, node) => {
+  let rowDrill = _.get('pagination.rows.drilldown', node)
+  let columnDrill = _.get('pagination.columns.drilldown', node)
+  let expandedColumns = _.get('pagination.columns.expanded', node)
+
+  extend(node, {
+    pagination: {
+      columns: {
+        drilldown: columnDrill && [],
+        skip: [],
+        expanded: expandedColumns,
+      },
+      rows: {
+        drilldown: rowDrill && [],
+        skip: [],
+        expanded: [],
+      },
+    },
+  })
+}
+
+export default {
+  validate: context =>
+    _.every(
+      ({ type, ranges, percents }) =>
+        (type !== 'numberRanges' && type !== 'percentiles') ||
+        (type === 'numberRanges' && ranges.length > 0) ||
+        (type === 'percentiles' && percents.length > 0),
+      _.concat(context.columns, context.rows)
+    ),
+  reactors: {
+    columns: 'self',
+    rows: 'self',
+    values: 'self',
+    pagination: 'self',
+    filters: 'others',
+    sort: 'self',
+  },
+  defaults: {
+    columns: [],
+    rows: [],
+    values: [],
+    filters: [],
+    sort: {},
+    pagination: {
+      columns: {
+        drilldown: null,
+        skip: [],
+        expanded: [],
+      },
+      rows: {
+        drilldown: null,
+        skip: [],
+        expanded: [],
+      },
+    },
+    showCounts: false,
+    context: {
+      results: [],
+    },
+  },
+  onDispatch(event, extend) {
+    let { type, node, previous, value } = event
+
+    if (type !== 'mutate') return
+
+    if (_.has('pagination.columns', value) || _.has('pagination.rows', value))
+      // if mutation is a pagination
+      addDrilldownToExpanded(extend, node, previous, value)
+    else if (_.has('sort', value)) {
+      // if sorting is changed we are preserving expanded columns
+      addDrilldownToExpanded(extend, node, previous, value)
+      resetExpandedRows(extend, node)
+    }
+    // if node configuration is changed resetting drilldowns
+    else resetPagination(extend, node, previous)
+  },
+  // Resetting the pagination when the tree is changed
+  // allows to return expected root results instead of nested drilldown
+  // EX: criteria filters didn't work properly when drilldown was applied
+  onUpdateByOthers(node, extend) {
+    resetPagination(extend, node, node)
+  },
+  shouldMergeResponse: node =>
+    // checking for presence of drilldown, skip, expanded in pagination
+    someNotEmpty(
```

Comment:
comment here mentions `expanded` but I don't see it as part of the check

Reply:
There appears to be an expand method in [src/exampleTypes/pivot.js](https://github.com/smartprocure/contexture-client/pull/186/files#diff-7135becdbd16d012b9e6f46a692c49b112fadcfe849d74f8a8610c482496eca1), uncertain if it's related to the same item.

Reply:
Yeah, this structure is changed entirely now. Removed the comment